src/main/java/com/ericsson/ei/erqueryservice/ERQueryService.java

```diff
@@ -60,97 +62,68 @@ public void setHttpRequest(HttpRequest request) {
-     * conditions.
-     *
-     * @param eventId
-     *            the id of the event.
-     * @param searchOption
-     *            the SearchOption to indicate whether to search up, down or
-     *            both ways from the eventId.
-     * @param limit
-     *            sets the limit of how many events up and/or down stream from
-     *            the eventId to include in the result.
-     * @param levels
-     *            sets the limit of how many levels up and/or down stream from
-     *            the eventId to include in the result.
-     * @param tree
-     *            whether or not to retain the tree structure in the result.
-     * @return ResponseEntity
-     */
-    public ResponseEntity getEventStreamDataById(String eventId, SearchOption searchOption, int limit,
-            int levels, boolean tree) {
-
-        String uri = null;
-        try {
-            if(StringUtils.isNotBlank(erBaseUrl)) {
-                // Request Body parameters
-                final SearchParameters searchParameters = getSearchParameters(searchOption);
-                request
-                        .setHttpMethod(HttpMethod.POST)
-                        .setBaseUrl(erBaseUrl)
-                        .setEndpoint(eventId)
-                        .addParam("limit", Integer.toString(limit))
-                        .addParam("levels", Integer.toString(levels))
-                        .addParam("tree", Boolean.toString(tree))
-                        .setBody(searchParameters.getAsJsonString(), ContentType.APPLICATION_JSON);
-
-                uri = request.getURI().toString();
-                LOGGER.debug("The URL to ER is: {}", uri);
-
-                return request.performRequest();
-            } else {
-                LOGGER.info("The URL to ER is not provided");
-            }
-        }
-        catch (Exception e) {
-            LOGGER.error("Error occurred while executing REST POST to {}, stacktrace: {}", uri, e);
+    private ResponseEntity sendRequestToER(String eventId, SearchOption searchOption, int limit,
+            int levels, boolean tree) throws IOException, URISyntaxException,
+            ClientProtocolException, PropertyNotFoundException {
+        if (StringUtils.isBlank(erBaseUrl)) {
+            throw new PropertyNotFoundException("The URL to ER is not provided");
         }
 
-        return null;
+        prepareRequest(eventId, searchOption, limit, levels, tree);
+        ResponseEntity response = sendRequest();
+        return response;
+    }
+
+    private void prepareRequest(String eventId, SearchOption searchOption, int limit,
+            int levels, boolean tree) throws IOException, URISyntaxException {
+        final SearchParameters searchParameters = getSearchParameters(searchOption);
+        request
+               .setHttpMethod(HttpMethod.POST)
+               .setBaseUrl(erBaseUrl)
+               .setEndpoint(eventId)
```

Comment:
isn't "/search" entrypoint part missing here ? (or what the entrypoint is named in ER)

Reply:
The url is taken from er.url property which probably contains the /search part

Reply:
But then its not a base url,, a base url is just the url to the service. and then a context path can be added otionally to the base url. All url paths after service url and context path is called entrypoints. If other entrypoint will be used in EI in future the "er.url" property can be reused and another property need to be added, and we weill have almost duplicated properties that contain same thing. If entrypoint is defined/handled in code, then "er.url" property can be reused for serveral entrypoints and usage in EI.